src/parser/druid/restoration/normalizers/ClearcastingNormalizer.ts

```diff
@@ -1,16 +1,16 @@
 import EventsNormalizer from 'parser/core/EventsNormalizer';
 
 import SPELLS from 'common/SPELLS';
-import { EventType } from 'parser/core/Events';
+import { AnyEvent, EventType } from 'parser/core/Events';
 
 const MAX_DELAY = 50;
 
 // Clearcasting buff fades before the regrowth cast event that consumed it shows up.
 // This swaps the order so the buff always fades AFTER the regrowth cast event.
 class ClearcastingNormalizer extends EventsNormalizer {
 
-  normalize(events) {
-    const fixedEvents = [];
+  normalize(events: AnyEvent[]): AnyEvent[] {
+    const fixedEvents: AnyEvent[] = [];
```

Comment:
Why is this function typed twice with AnyEvent[]

Reply:
I don't know what you mean by `typed twice with AnyEvent[]`
Since the events are typed to AnyEvent[] then the return of the function is typed to it. 
Its how it is done in mistweaver so i just imported that. If you don't want the function to indicate its return I can remove it

Reply:
I see what you're saying now but still up to you if you want it removed

Reply:
It's a bit redundant when you're 100% aware that you're taking in AnyEvent array and just returning the same. 
It'd be interesting if we were transforming from one event to another probably, but in the case of AnyEvent it seems redundant - can stay or not, I don't think I'd write it myself if I were to write the function but it causes no harm.

Reply:
by explicitly setting the return type, you'll get an typescript error if you attempt to return something else by accident. however because it overrides the EventsNormalizer abstract function it wont let you try to return something invalid anyway.

in any case its stylistic /shrug

Reply:
I literally copy and pasted that part from mw stuff which i don't fully remember if it was Anom or me who did the normalizers to typescript

Reply:
So /shrug i can fix